grabbit/src/main/groovy/com/twcable/grabbit/server/batch/steps/jcrnodes/JcrNodesProcessor.groovy

```diff
@@ -88,10 +86,57 @@ class JcrNodesProcessor implements ItemProcessor<JcrNode, Node> {
             }
         }
 
-        final List<JcrProperty> properties = jcrNode.properties.toList()
-        Node.Builder nodeBuilder = Node.newBuilder()
+        // Skip the node as it already been processed by parent node
+        if(isRequiredNode(jcrNode)) {
+            return null;
+        } else {
+            // Build parent node
+            return buildNode(jcrNode, getRequiredChildNodes(jcrNode))
```

Comment:
```groovy
final requiredChildNodes = hasMandatoryChildNodes(jcrNode) ? getRequiredChildNodes(jcrNode) : null
return buildNode(jcrNode, requiredChildNodes)
...
private boolean hasMandatoryChildNodes(jcrNode) {
    jcrNode.getPrimaryNodeType().getChildNodeDefinitions().find { it.mandatory }
}
```
so we don't have to iterate over all of a node's child nodes if we don't have to (as is the case most of the time)